### DISCUSSION TITLE
Datatable filters - menu mode for filtering options

### DISCUSSION THREAD

**Discussion by james-gulland** [primefaces/community#2151] [PrimeVue]
Hi,

I am using the PrimeVue datatable, and I had a few questions about customising when filterDisplay="menu".

1) Is it possible to not show the FilterOperator field (Matches ALL or ANY) - this is fairly useless field for users but I cant just remove it from the code otherwise the filters wont work.  Is there a way of disabling in the UI (and defaulting to ALL)?

2) I have the FilterOperator, FilterMatchMode and also an InputText field in the menu. Is there a way of focusing the field on the InputText by default rather than the first field?  So a user can start typing immediately

3) Is there a way to enable pressing of the Enter key to apply the filter from the menu?

Thanks! I really love these components BTW :)

**Comment by LewisDamy** [comment]
Hi, I've just had a very similar issue. In my case it was that I want to hardcode it the option of the FilterOperator while allowing the user tho enter a text i nthe InputText field. The solution I've came up has been to overwrite the filter for each field and customize it as I want. 

For instance taking the StackBlitz example for the LazyTable. The table has this structure:

```vue
<DataTable
      :value="customers"
      lazy
      paginator
      removableSort
      :first="lazyParams.first"
      :rows="lazyParams.rows"
      :totalRecords="totalRecords"
      v-model:filters="filters"
      ref="dt"
      dataKey="id"
      :loading="loading"
      filterDisplay="row"
      @page="onPage"
      @sort="onSort"
      @filter="onFilter"
    >
    <Column field="name" :showFilterMenu="false" header="Name" sortable>
        <!-- Custom filter input and action button -->
        <template #filter="{ filterCallback, filterModel }">
          <div class="flex items-center gap-2">
            <InputText
              v-model="filterModel.value"
              class="p-column-filter"
              placeholder="Search by name"
            />
            <button class="p-button" @click="filterCallback()">
              <span class="pi pi-filter"></span>
            </button>
          </div>
        </template>

        <!-- Custom filter clear button -->
        <template #filterclear="{ filterModel, filterCallback }">
          <button
            class="p-button p-button-sm p-button-secondary"
            @click="
              filterModel.value = null;
              filterCallback();
            "
          >
            <span class="pi pi-times"></span>
          </button>
        </template>
      </Column>
</DataTable>
```
In order to overwrite it. You must add the `:showFilterMenu="false"` and then add yours. In my case I've added the `<template #filter>` which contains the input and the submit button, and I also added the `<template #filterclear>`. Note that for both templates it can be access the functions `filterModel` and `filterCallback` to perform the action/behaviour that you'd like to have (clear/run filter etc).

The result is this:
Before filtering:
![image](https://github.com/user-attachments/assets/f5ccf7f0-12fa-423c-997c-0e80cede2b86)
After adding a text and running:
![image](https://github.com/user-attachments/assets/18c47a2d-c3f9-4eec-99f8-793d6f9e63e6)
And I can filter with the filter button, without issues. In my case I've used the lazy layout since my code will handle a large amount of data hence, this approach.

One remainder is to be extra carful regarding the version of the libraries, specially primevue and related.

Here is an live action implementation of the solution at Stackblitz:
https://stackblitz.com/edit/xmdjekhu?file=src%2FApp.vue
